src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,121 @@
+package duke.main;
+
+import duke.exception.EmptyTaskException;
+import duke.exception.InvalidCommandException;
+import duke.exception.InvalidDateException;
+import duke.exception.InvalidIndexException;
+import duke.exception.MissingDateException;
+import duke.task.Task;
+
+/**
+ * Parser is a class used by Duke to interpret user inputs as commands.
+ */
+public class Parser {
+    private final TaskList taskList;
+    private StringBuilder reply = new StringBuilder();
+
+    Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Interprets user input and executes commands.
+     * The method supports commands for adding, deleting, viewing and completing tasks.
+     * If invalid input is given, users are informed and told to try again.
+     *
+     * @param input User command in console.
+     * @return Output message upon execution of command.
+     */
+    String parse(String input) {
+        try {
+            if (input.equals("list")) {
+                list();
+            } else if (input.startsWith("done")) {
+                done(input);
+            } else if (input.startsWith("delete")) {
+                delete(input);
+            } else if (input.startsWith("find")) {
+                find(input);
+            } else {
+                add(input);
+            }
```

Comment:
This usage of if - else works for sure, but perhaps the use of switch - cases would be more appropriate in this case.
Minor issue tho xD

Reply:
Thanks for the suggestion! I forgot I could simply extract the first word with a scanner instead of using `String.startsWith()`. Not that it matters now that we're using GUI, but at least I've found my own way of doing so.